WARNING! This patch is buggy!
It fixes compilation without gtkspell and breaks compilation with gtkspell.
Index: balsa-2.4.0/src/Makefile.am
===================================================================
--- balsa-2.4.0.orig/src/Makefile.am
+++ balsa-2.4.0/src/Makefile.am
@@ -87,13 +87,13 @@ balsa_print_source = print-gtk.c	\
 	balsa-print-object-text.h
 
 if BUILD_WITH_GTKSPELL
-balsa_gtkspell_extra =
-balsa_gtkspell_extra_dist =	\
+balsa_gtkspell_extra_dist =
+balsa_gtkspell_extra =		\
 		spell-check.c		\
 		spell-check.h
 else
-balsa_gtkspell_extra_dist =
-balsa_gtkspell_extra =		\
+balsa_gtkspell_extra =
+balsa_gtkspell_extra_dist =	\
 		spell-check.c		\
 		spell-check.h
 endif
Index: balsa-2.4.0/src/sendmsg-window.c
===================================================================
--- balsa-2.4.0.orig/src/sendmsg-window.c
+++ balsa-2.4.0/src/sendmsg-window.c
@@ -366,7 +366,7 @@ static const GtkActionEntry entries[] =
      NULL, G_CALLBACK(insert_signature_cb)},
     {"QuoteMessages", NULL, N_("_Quote Message(s)"), NULL,
      NULL, G_CALLBACK(quote_messages_cb)},
-#if !HAVE_GTKSPELL
+#if HAVE_GTKSPELL
     {"CheckSpelling", GTK_STOCK_SPELL_CHECK, N_("C_heck Spelling"), NULL,
      N_("Check the spelling of the message"),
      G_CALLBACK(spell_check_cb)},
@@ -6559,7 +6559,6 @@ spell_check_menu_cb(GtkToggleAction * ac
         sw_spell_detach(bsmsg);
 }
 
-#else                           /* HAVE_GTKSPELL */
 /* spell_check_cb
  * 
  * Start the spell check
@@ -6605,6 +6604,7 @@ spell_check_cb(GtkAction * action, Balsa
     balsa_spell_check_start(sc, GTK_WINDOW(bsmsg->window));
 }
 
+#else                           /* HAVE_GTKSPELL */
 static void
 sw_spell_check_response(BalsaSpellCheck * spell_check, gint response, 
                         BalsaSendmsg * bsmsg)
